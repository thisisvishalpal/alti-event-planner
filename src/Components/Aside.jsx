import { useSelector, useDispatch } from "react-redux";
import { Navbar } from "react-bootstrap";
import { NavLink, useNavigate } from "react-router-dom";
import { logout } from "Slices";
import { urls } from "Utils";

const {
  root,
  connections,
  messages,
  notifications,
  search,
  settings,
  signIn,
  signUp,
} = urls;
export const Aside = () => {
  const data = useSelector(({ userAuth }) => userAuth);

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleLogout = () => {
    dispatch(logout());
    navigate(root);
  };
  return (
    <aside>
      <Navbar>
        <ul
          style={{
            listStyle: "none",
            padding: 0,
          }}
        >
          {/* <li style={{ marginBottom: "10px" }}>
            <NavLink to={signIn} className="nav-link">
              Signin
            </NavLink>
          </li> */}
          {/* <li style={{ marginBottom: "10px" }}>
            <NavLink to={signUp} className="nav-link">
              Signup
            </NavLink>
          </li> */}
          <li style={{ marginBottom: "10px" }}>
            <NavLink to={root} className="nav-link">
              Feeds
            </NavLink>
          </li>
          <li style={{ marginBottom: "10px" }}>
            <NavLink to={`user/${data?.username}`} className="nav-link">
              Profile
            </NavLink>
          </li>
          <li style={{ marginBottom: "10px" }}>
            <NavLink to={connections} className="nav-link">
              Connections
            </NavLink>
          </li>
          <li style={{ marginBottom: "10px" }}>
            <NavLink to={messages} className="nav-link">
              Messages
            </NavLink>
          </li>
          <li style={{ marginBottom: "10px" }}>
            <NavLink to={notifications} className="nav-link">
              Notifications
            </NavLink>
          </li>
          <li style={{ marginBottom: "10px" }}>
            <NavLink to={search} className="nav-link">
              Search
            </NavLink>
          </li>
          <li style={{ marginBottom: "10px" }}>
            <NavLink to={settings} className="nav-link">
              Settings
            </NavLink>
          </li>
          <hr />
          <li style={{ marginBottom: "10px" }}>
            <NavLink onClick={handleLogout} className="nav-link">
              Logout
            </NavLink>
          </li>
        </ul>
      </Navbar>
    </aside>
  );
};
