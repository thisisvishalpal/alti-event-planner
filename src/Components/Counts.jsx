export const Counts = () => {
  return <>hey there, this is count component</>;
};
