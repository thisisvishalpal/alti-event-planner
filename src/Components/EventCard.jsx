import React from "react";
import { Card, Button, Row, Col } from "react-bootstrap";

export const EventCard = ({ event, onSelect }) => {
  const { event_name, event_description, sell_price } = event.payload.items[0];

  const { image, event_date, event_time } = event.payload;

  return (
    <Col md={4} className="mb-4">
      <Card className="event-card shadow-sm border-0">
        <Card.Img
          variant="top"
          src={image}
          alt={event_name}
          className="event-image"
        />
        <Card.Body className="px-3 py-4">
          <Card.Title className="fw-bold text-dark">{event_name}</Card.Title>
          <Card.Text className="text-muted">{event_description}</Card.Text>

          {/* Event details in a row */}
          <Row>
            <Col xs={6} className="text-muted">
              <strong>Date:</strong> {event_date}
            </Col>
            <Col xs={6} className="text-muted text-end">
              <strong>Time:</strong> {event_time}
            </Col>
          </Row>
          <Row>
            <Col xs={6} className="text-muted">
              <strong>Location:</strong> {event.city.toUpperCase()}
            </Col>
            <Col xs={6} className="text-muted text-end">
              <strong>Price:</strong> {sell_price}
            </Col>
          </Row>

          <div className="text-center mt-3">
            <Button
              variant="primary"
              size="lg"
              onClick={() => onSelect(event)}
              className="w-100"
            >
              Select Seats
            </Button>
          </div>
        </Card.Body>
      </Card>
    </Col>
  );
};
