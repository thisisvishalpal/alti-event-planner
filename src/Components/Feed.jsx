import React, { useState } from "react";
import { Card, Row, Col, Button, Image, Form } from "react-bootstrap";

export const Feed = ({ posts }) => {
  const [likes, setLikes] = useState(posts.map(() => false));
  const [comments, setComments] = useState(posts.map(() => []));

  const handleLike = (index) => {
    setLikes((prev) => prev.map((liked, i) => (i === index ? !liked : liked)));
  };

  const handleAddComment = (index, commentText) => {
    setComments((prev) =>
      prev.map((commentList, i) =>
        i === index ? [...commentList, commentText] : commentList
      )
    );
  };

  return (
    <div className="feeds">
      <Row>
        {posts.map((post, index) => (
          <Col md={6} key={post.id} className="mb-4">
            <Card className="shadow-sm">
              <Card.Body>
                {/* Post Header */}
                <Row className="align-items-center">
                  <Col xs={2}>
                    <Image
                      src={post.user.profilePicture}
                      roundedCircle
                      className="img-fluid"
                    />
                  </Col>
                  <Col>
                    <h5 className="mb-0">{post.user.name}</h5>
                    <small className="text-muted">{post.date}</small>
                  </Col>
                </Row>

                {/* Post Content */}
                <Card.Text className="mt-3">{post.content}</Card.Text>
                {post.image && (
                  <Image
                    src={post.image}
                    alt="Post"
                    className="img-fluid rounded"
                  />
                )}

                {/* Post Actions */}
                <div className="mt-3 d-flex justify-content-between">
                  <Button
                    variant={likes[index] ? "primary" : "outline-primary"}
                    onClick={() => handleLike(index)}
                  >
                    {likes[index] ? "Liked" : "Like"}
                  </Button>
                  <Button variant="outline-secondary">Share</Button>
                </div>

                {/* Comments Section */}
                <div className="mt-4">
                  <h6>Comments</h6>
                  <ul className="list-unstyled">
                    {comments[index].map((comment, i) => (
                      <li key={i} className="mb-2">
                        <strong>{comment.user}:</strong> {comment.text}
                      </li>
                    ))}
                  </ul>

                  {/* Add Comment Form */}
                  <Form
                    onSubmit={(e) => {
                      e.preventDefault();
                      const form = e.target;
                      const commentText = form.elements.comment.value.trim();
                      if (commentText) {
                        handleAddComment(index, {
                          user: "You",
                          text: commentText,
                        });
                        form.reset();
                      }
                    }}
                  >
                    <Form.Group controlId={`comment-${index}`}>
                      <Form.Control
                        type="text"
                        placeholder="Add a comment..."
                        name="comment"
                      />
                    </Form.Group>
                  </Form>
                </div>
              </Card.Body>
            </Card>
          </Col>
        ))}
      </Row>
    </div>
  );
};
