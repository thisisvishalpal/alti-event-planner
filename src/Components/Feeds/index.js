export { UploadSection } from "./UploadSection";
export { FeedSection } from "./Feed";
