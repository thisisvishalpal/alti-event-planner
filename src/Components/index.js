export { ThemeToggler } from "./ThemeToggler";
export {
  UserPosts,
  ActionButton,
  ProfileUsername,
  ProfileTabs,
} from "./Profile";
export { UploadSection, FeedSection } from "./Feeds";
export {
  Aside,
  RightAside,
  Header,
  OffcanvasExample,
  Footer,
  Layout,
} from "./Layout";
export { SearchBar } from "./SearchBar";
export { UserRow } from "./UserRow";
export { Navigations } from "./Navigations";
export { Spinner } from "./Spinner";
export { SpinnerTwo } from "./SpinnerTwo";
export { ConnectionTabs } from "./Connections";
export {
  UsernameEmailForm,
  PersonalInfoForm,
  CareerInformationForm,
  SecurityForm,
  SearchFilterForm,
  SearchTypeSelector,
  LanguageForm,
  LanguageToggler,
} from "./FormInput";
export { PostContent, PostHeader, PostModal, LikeCommentShare } from "./Post";
export { ShowMessage, ShareButton } from "./Generic";
