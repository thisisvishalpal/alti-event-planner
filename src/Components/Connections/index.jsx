import React, { useState } from "react";
import { Tab, Tabs, Container } from "react-bootstrap";

import { ShowMessage, UserRow } from "Components";

export const ConnectionTabs = ({ state, followers = [], following = [] }) => {
  const [activeTab, setActiveTab] = useState(state || "followers");

  const handleTabSelect = (tab) => setActiveTab(tab);

  // Render tab content based on user data
  const renderTabContent = (users) => {
    if (!users.length) {
      return (
        <ShowMessage
          heading="No users found."
          secondary="Try connecting with new users."
        />
      );
    }
    return users?.map((user) => <UserRow key={user._id} user={user} />);
  };

  return (
    <Container className="mt-4">
      <Tabs activeKey={activeTab} onSelect={handleTabSelect} justify>
        <Tab eventKey="followers" title={`Followers ${followers?.length}`}>
          {renderTabContent(followers)}
        </Tab>

        <Tab eventKey="following" title={`Following ${following?.length}`}>
          {renderTabContent(following)}
        </Tab>
      </Tabs>
    </Container>
  );
};
