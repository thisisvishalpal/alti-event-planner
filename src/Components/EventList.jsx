import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import { EventCard, SeatSelectionModal } from "Components";
import { fetchEvents } from "Services";

export const EventList = () => {
  const [selectedEvent, setSelectedEvent] = useState(null);
  const [showModal, setShowModal] = useState(false);
  const dispatch = useDispatch();

  const { events, error, loading } = useSelector((state) => state.allEvents);

  useEffect(() => {
    dispatch(fetchEvents()); // Fetch data on mount
  }, [dispatch]);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error: {error}</p>;

  const handleSelectEvent = (event) => {
    setSelectedEvent(event);
    setShowModal(true);
  };

  const handleCloseModal = () => {
    setShowModal(false);
    setSelectedEvent(null); // Clear the selected event when the modal is closed
  };

  return (
    <div className="container mt-5">
      <div className="row">
        {events.length ? (
          events?.map((event) => (
            <EventCard
              key={event.id}
              event={event}
              onSelect={handleSelectEvent}
            />
          ))
        ) : (
          <h3 style={{ textAlign: "center" }}>Please try after sometime</h3>
        )}
      </div>

      {/* Show the modal if an event is selected */}
      {selectedEvent && (
        <SeatSelectionModal
          event={selectedEvent}
          show={showModal}
          onClose={handleCloseModal}
        />
      )}
    </div>
  );
};
