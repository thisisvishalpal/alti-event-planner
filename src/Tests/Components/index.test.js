import React from "react";
import { render } from "@testing-library/react";
import { SwitchSelector, Footer, EventCard } from "Components";

describe("SwitchSelector Component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <SwitchSelector
        labelLeft="Light"
        labelRight="Dark"
        onChange={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Footer Component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<Footer />);

    // Compare the rendered output to the saved snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
