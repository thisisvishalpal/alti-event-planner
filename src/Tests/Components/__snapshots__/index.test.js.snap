// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <hr />
    <div
      class="footer-container"
    >
      <div
        class="footer-section"
      >
        <h3>
          About Event planner
        </h3>
        <p>
          Your one-stop platform to explore events in various fields.
        </p>
      </div>
      <div
        class="footer-section"
      >
        <h3>
          Quick Links
        </h3>
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h3>
          Follow Us
        </h3>
        <ul
          class="social-links"
        >
          <li>
            <a
              href="#"
            >
              Facebook
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Twitter
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              LinkedIn
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      <p>
        Â© 2024 Event Planner. All Rights Reserved.
      </p>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`SwitchSelector Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="switch-container"
  >
    <span
      class="label"
    >
      Light
    </span>
    <div
      class="switch "
    >
      <div
        class="slider"
      />
    </div>
    <span
      class="label"
    >
      Dark
    </span>
  </div>
</DocumentFragment>
`;
